include/seqan3/core/type_list/traits.hpp

```diff
@@ -132,6 +145,27 @@ inline constexpr ptrdiff_t count =  (SEQAN3_IS_SAME(query_t, pack_t) + ... + 0);
 template <typename query_t, typename ...pack_t>
 inline constexpr ptrdiff_t find = seqan3::pack_traits::detail::find<query_t, pack_t...>();
 
+/*!\brief Get the index of the first type in a pack that satisfies the given predicate.
+ * \tparam pred_t   The type you are searching for.
+ * \tparam pack_t   The type pack.
+ * \returns The position or `-1` if no types match.
+ * \ingroup type_list
+ *
+ * \details
+ *
+ * Note that the predicate must be given as a type template (variable templates cannot be passed as template arguments
+ * unfortunately). This means e.g. `find_if<std::is_integral, float, double, int, float>` (not `std::is_integral_v`!).
+ *
+ * ### (Compile-time) Complexity
+ *
+ * * Number of template instantiations: O(n), possibly == `i`, where `i` is the return value
+ * * Other operations: O(n), possibly == `i`, where `i` is the return value
+ *
+ *  Only the predicate is instantiated.
+ */
+template <template <typename> typename pred_t, typename ...pack_t>
+inline constexpr ptrdiff_t find_if = seqan3::pack_traits::detail::find_if<pred_t, pack_t...>();
```

Comment:
Can I ask what the difference is between this and the actual definition earlier?

Reply:
this is variable (template), the above is a function (template)!